src/normalizeUnit.ts

```diff
@@ -0,0 +1,20 @@
+export function normalizeUnit(prevUnit: string, currentUnit: string, value: number): number {
+    for (const units of SUPPORTED_UNITS) {
+        const prevUnitIndex = units.indexOf(prevUnit);
+        const currentUnitIndex = units.indexOf(currentUnit);
+
+        if (prevUnitIndex >= 0 && currentUnitIndex >= 0) {
+            const unitDiff = prevUnitIndex - currentUnitIndex;
+
+            return value * UNIT_CONVERSION_MULTIPLIER ** unitDiff;
+        }
+    }
+
+    return value;
+}
+
+const UNIT_CONVERSION_MULTIPLIER = 1000;
+const TIME_UNITS = ['s', 'ms', 'us', 'ns'];
+const ITER_UNITS = TIME_UNITS.map((unit) => `${unit}/iter`);
+const OPS_PER_TIME_UNIT = [...TIME_UNITS].reverse().map((unit) => `ops/${unit}`);
```

Comment:
giga-nitpick: I noticed you have used two different cloning techniques on the same PR, maybe is better for consistency to pick one.

1. [...previousValue]
2. previousValue.slice()